### PR TITLE
Fix kernel identify shader on Metal.

### LINKED ISSUE #120: Video Preview in "No filter" mode fails when Metal is active and Scale Media to Fill is active
### Steps to reproduce:

 - Start the demo app
 -  Disable Camera Open GL and Camera Open GL Capture
 -  Enable Camera Metal and Metal Filters
 -  Enable Scale Media To Fill
 - Tap on Start
 - Tap and hold on the shutter button to record a video
 - Release the button
 - Check that video preview is blank/black
 - If you tap on the filter option and switch to another filter type it works
 - but if you go back to no filter it goes black again.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #120

See: https://jira.tumblr.net/browse/PROD-15637

### Changes

This PR changes the variable types on the kernelIdentity shader in order for this check to work properly:

`(inWidth - outWidth) > 0`

Using `uint` will make this comparation to always be true even when outWidth is larger than inWidth.
It will also cause the offsset to overflow and be a large number and cause the bug we observe.

Up until now we have been *lucky* that the outTexture was always smaller, and that is also why the bug only manifested itself on certain hardware depending of the screen sizes and video resolution calculations.

### How to test

 - Start the demo app on a real device. On my tests I need to use an iPhone 12 Pro to see the original bug.
 -  Disable Camera Open GL and Camera Open GL Capture
 -  Enable Camera Metal and Metal Filters
 -  Enable Scale Media To Fill
 - Tap on Start
 - Tap and hold on the shutter button to record a video
 - Release the button
 - Check that video preview is working properly

### CODE REVIEW

**Review by danielebogo — APPROVED** [review]
LGTM!
Waiting for @bjtitus feedback and see if this might affect WP too.

**Comment by bjtitus** [comment]
✅ Verified on an iPhone 10.

This might unblock us from using Metal on WPiOS!

**Review by bjtitus — APPROVED** [review]

**Pull request merged**
